client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/utils/BsonHelper.java

```diff
@@ -0,0 +1,136 @@
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Set<String> documentKeys(Object document) {
+        try {
+            Class<?> documentClass = Class.forName("org.bson.Document");
+            return (Set<String>) documentClass.getMethod("keySet").invoke(document);
+        } catch (ClassNotFoundException | IllegalAccessException | InvocationTargetException |
+                 NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Boolean isDocument(Object value) {
+        Class<?> documentClass;
+        try {
+            documentClass = Class.forName("org.bson.Document");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        return documentClass.isInstance(value);
+    }
+
+    public static Object convertToQueryDocument(Object query) {
+        try {
+            Method toBsonDocument = query.getClass().getMethod("toBsonDocument");
+            Object bsonDocument = toBsonDocument.invoke(query);
+
+            Object documentCodec = documentCodecClass().getDeclaredConstructor().newInstance();
+            Method asBsonReader = bsonDocumentClass().getMethod("asBsonReader");
+            Method builder = decoderContextClass().getMethod("builder");
+            Object builderInstance = builder.invoke(null);
+            Method build = builderInstance.getClass().getMethod("build");
+
+            Method decode = documentCodecClass().getMethod("decode", bsonReaderClass(), decoderContextClass());
+            return decode.invoke(documentCodec, asBsonReader.invoke(bsonDocument), build.invoke(builderInstance));
+        } catch (NoSuchMethodException | InstantiationException | IllegalAccessException |
+                 InvocationTargetException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String getType(Object bsonType) {
+        try {
+            Class<?> bsonTypeClassMapClass = Class.forName("org.bson.codecs.BsonTypeClassMap");
+            Class<?> bsonTypeClass = Class.forName("org.bson.BsonType");
+            Object bsonTypeClassMap = bsonTypeClassMapClass.getDeclaredConstructor().newInstance();
+            Method get = bsonTypeClassMapClass.getMethod("get", bsonTypeClass);
+            Object type = get.invoke(bsonTypeClassMap, bsonType);
+            return (String) type.getClass().getMethod("getTypeName").invoke(type, null);
+        } catch (ClassNotFoundException | NoSuchMethodException | InstantiationException | IllegalAccessException |
+                 InvocationTargetException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Object getTypeFromNumber(Integer number) {
+        Class<?> bsonTypeClass;
+        try {
+            bsonTypeClass = Class.forName("org.bson.BsonType");
```

Comment:
hmm... so here we would have a reference to an object from MongoDB... so `Class.forName` should be fine (this applies for previous comments as well). however, something like this will not work under `instrumentation` module (again, long story, might be easier to explain in a meeting). amyway, you will see it immediately when you run with both embedded and external drivers